src/kernels/jupyter/launcher/jupyterConnection.node.ts

```diff
@@ -149,9 +149,11 @@ export class JupyterConnectionWaiter implements IDisposable {
                 return;
             }
 
+            // For more recent versions of Jupyter the web pages are served from `/tree` and the api is at the root.
+            const pathName = url.pathname.endsWith('/tree') ? url.pathname.replace('/tree', '') : url.pathname;
```

Comment:
Looks like we have a similar approach when working with jupyter hub as well, we strip out `tree`, as the ws & rest end points are at the root level.